### DISCUSSION TITLE
State type not working in Getter

### DISCUSSION THREAD

**Discussion by nVitius** [vuejs/pinia#677] [Help and Questions] [answered]
I ran into a TS issue when trying to create a getter. Not sure if this might be environment-related though; I wanted to ask here to see if someone else has seen this.

My store looks like this:
```typescript
type CoreStoreState = {
  users: UsersQuery['users']
}

export const useCoreStore = defineStore('core', {
  state: (): CoreStoreState => ({ users: [] }),
  getters: {
    someGetter: (st) => {
      ...
    }
  }
}
```

Trying to autocomplete on `st` does not resolve the type correctly:
![image](https://user-images.githubusercontent.com/3804041/133380399-366e1af1-a233-43a9-af17-fd8e7160ff7d.png)

I'm using TS 4.4.3 with Intellij IDEA 2021.2.1

**Comment by nVitius** [comment, marked as answer]
This was definitely an IDE issue. Restarting the TypeScript service fixed this:
![image](https://user-images.githubusercontent.com/3804041/133383037-40526273-a1e4-4fee-b868-ec1d5477a7de.png)

Sorry for the noise.

**Comment by theoephraim** [comment]
FYI, you can also type the items in the initial state directly, for example:
```
export const useUsersStore = defineStore('users', {
  state: () => ({
    users: [] as User[],
    someUninitializedString: null as null | string,
  }),
}
```
